internal/client/block_rules.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2025 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package client
+
+import "github.com/ChainSafe/gossamer/internal/primitives/runtime"
+
+type BadBlockData[H runtime.Hash, N runtime.Number] struct {
+	number N
+	hash   H
```

Comment:
How will they set the attributes if they are private?
```suggestion
	Number N
	Hash   H
```